server/src/main/java/org/apache/cloudstack/vm/schedule/VMScheduleManagerImpl.java

```diff
@@ -0,0 +1,322 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cloudstack.vm.schedule;
+
+import com.cloud.api.query.MutualExclusiveIdsManagerBase;
+import com.cloud.event.ActionEvent;
+import com.cloud.event.EventTypes;
+import com.cloud.exception.InvalidParameterValueException;
+import com.cloud.user.AccountManager;
+import com.cloud.utils.DateUtil;
+import com.cloud.utils.Pair;
+import com.cloud.utils.component.PluggableService;
+import com.cloud.utils.db.Filter;
+import com.cloud.utils.db.SearchBuilder;
+import com.cloud.utils.db.SearchCriteria;
+import com.cloud.utils.db.Transaction;
+import com.cloud.utils.db.TransactionCallback;
+import com.cloud.utils.exception.CloudRuntimeException;
+import com.cloud.vm.VirtualMachine;
+import com.cloud.vm.VirtualMachineManager;
+import org.apache.cloudstack.api.command.user.vm.CreateVMScheduleCmd;
+import org.apache.cloudstack.api.command.user.vm.DeleteVMScheduleCmd;
+import org.apache.cloudstack.api.command.user.vm.ListVMScheduleCmd;
+import org.apache.cloudstack.api.command.user.vm.UpdateVMScheduleCmd;
+import org.apache.cloudstack.api.response.ListResponse;
+import org.apache.cloudstack.api.response.VMScheduleResponse;
+import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.framework.messagebus.MessageBus;
+import org.apache.cloudstack.vm.schedule.dao.VMScheduleDao;
+import org.apache.log4j.Logger;
+import org.springframework.scheduling.support.CronExpression;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.TimeZone;
+
+public class VMScheduleManagerImpl extends MutualExclusiveIdsManagerBase implements VMScheduleManager, PluggableService {
+
+    private static Logger LOGGER = Logger.getLogger(VMScheduleManagerImpl.class);
+
+    @Inject
+    private VMScheduleDao vmScheduleDao;
+    @Inject
+    private VirtualMachineManager virtualMachineManager;
+    @Inject
+    private VMScheduler vmScheduler;
+    @Inject
+    private MessageBus messageBus;
+    @Inject
+    private AccountManager accountManager;
+
+    @Override
+    public List<Class<?>> getCommands() {
+        final List<Class<?>> cmdList = new ArrayList<>();
+        cmdList.add(CreateVMScheduleCmd.class);
+        cmdList.add(ListVMScheduleCmd.class);
+        cmdList.add(UpdateVMScheduleCmd.class);
+        cmdList.add(DeleteVMScheduleCmd.class);
+        return cmdList;
+    }
+
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_VM_SCHEDULE_CREATE, eventDescription = "Creating VM Schedule")
+    public VMScheduleResponse createSchedule(CreateVMScheduleCmd cmd) {
+        VirtualMachine vm = virtualMachineManager.findById(cmd.getVmId());
```

Comment:
This will give any VM and not just user vm.
I was able to create a job for SSVM
```
(local) 🦅 > create vmschedule name=gfdgfd description=gfdgfd schedule="10 5 * * 1" timezone=UTC action=STOP virtualmachineid=b2bd4347-6157-485e-9422-936a509908de startdate="2023-04-21 20:10:10" enabled=true
{
  "vmschedule": {
    "action": "STOP",
    "description": "gfdgfd",
    "enabled": true,
    "id": "b1972d7e-62f2-4f01-854b-d89b6cc7fae5",
    "name": "gfdgfd",
    "schedule": "10 5 * * 1",
    "startdate": "2023-04-21T20:10:10+0530",
    "timezone": "UTC",
    "virtualmachineid": "b2bd4347-6157-485e-9422-936a509908de"
  }
}
(local) 🐒 > list systemvms id=b2bd4347-6157-485e-9422-936a509908de
{
  "count": 1,
  "systemvm": [
    {
      "agentstate": "Up",
      "created": "2023-04-21T18:27:27+0530",
      "dns1": "10.147.28.6",
      "gateway": "192.168.2.1",
      "hasannotations": false,
      "hostcontrolstate": "Enabled",
      "hostid": "34bb6388-9674-45e6-a1c2-21c380cfc8d7",
      "hostname": "SimulatedAgent.288cd369-c7ff-46a5-924b-abe17d18c6a4",
      "hypervisor": "Simulator",
      "id": "b2bd4347-6157-485e-9422-936a509908de",
      "isdynamicallyscalable": false,
      "linklocalip": "169.254.224.254",
      "linklocalmacaddress": "0e:00:a9:fe:e0:fe",
      "linklocalnetmask": "255.255.0.0",
      "name": "s-1-VM",
      "podid": "dbfef4bd-ceb5-4d64-96fc-d9b014e42014",
      "podname": "POD0",
      "privateip": "172.16.15.147",
      "privatemacaddress": "1e:00:23:00:00:92",
      "privatenetmask": "255.255.255.0",
      "publicip": "192.168.2.2",
      "publicmacaddress": "1e:00:36:00:00:c8",
      "publicnetmask": "255.255.255.0",
      "state": "Running",
      "systemvmtype": "secondarystoragevm",
      "templateid": "00715843-e043-11ed-abc1-645d8651f45a",
      "templatename": "SystemVM Template (simulator)",
      "version": "4.19.0.0-SNAPSHOT",
      "zoneid": "f80d6c12-a17d-4ca9-bace-c5e4b1db6108",
      "zonename": "Sandbox-simulator"
    }
  ]
}
```
Use UserVmDao or Manager unless it is meant purposefully to allow all VMs